From a3ec177ccee07a559659a5eeca7dcd9d55df1aae Mon Sep 17 00:00:00 2001
From: Jonas Karlman <jonas@kwiboo.se>
Date: Sat, 16 Jun 2018 14:20:43 +0200
Subject: [PATCH] LinuxRendererGLES: remove early exit from Flush

Fixes an issue with unreleased video buffers when AddVideoPicture is called between Flush and ValidateRenderTarget
Also log about unreleased video buffers in AddVideoPicture to help catch potential future issues
---
 xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp b/xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp
index 067199d88059..e721db639fff 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGLES.cpp
@@ -203,6 +203,11 @@ int CLinuxRendererGLES::NextYV12Texture()
 void CLinuxRendererGLES::AddVideoPicture(const VideoPicture &picture, int index, double currentClock)
 {
   CPictureBuffer &buf = m_buffers[index];
+  if (buf.videoBuffer)
+  {
+    CLog::LogF(LOGERROR, "unreleased video buffer");
+    buf.videoBuffer->Release();
+  }
   buf.videoBuffer = picture.videoBuffer;
   buf.videoBuffer->Acquire();
   buf.loaded = false;
@@ -344,9 +349,6 @@ void CLinuxRendererGLES::LoadPlane(YUVPLANE& plane, int type,
 
 void CLinuxRendererGLES::Flush()
 {
-  if (!m_bValidated)
-    return;
-
   glFinish();
 
   for (int i = 0 ; i < m_NumYV12Buffers ; i++)
